lib/Ninja/ManifestLoader.cpp

```diff
@@ -395,11 +395,12 @@ class ManifestLoaderImpl: public ParseActions {
       
     // FIXME: Mange recursive lookup? Ninja crashes on it.
       
-    // Support "in" and "out".
-    if (name == "in") {
+    // Support "in", "in_newline" and "out".
+    if (name == "in" || name == "in_newline") {
+      const auto separator = name == "in" ? ' ' : '\n';
       for (unsigned i = 0, ie = decl->getNumExplicitInputs(); i != ie; ++i) {
         if (i != 0)
-          result << " ";
+          result << separator;
```

Comment:
I have a mild preference for just inlining the `separator` variable, so reader doesn't have to ever look an see if it is used elsewhere, but not a big deal.

Reply:
Are you suggesting something like this?
```suggestion
          result << (name == "in" ? ' ' : '\n');
```

It seems less readable to me :)